sbi/neural_nets/embedding_nets.py

```diff
@@ -44,81 +45,151 @@ def forward(self, x: Tensor) -> Tensor:
         return self.net(x)
 
 
-class CNNEmbedding(nn.Module):
+def get_new_cnn_output_size(
+    input_dim: int,
+    conv_layer: Union[nn.Conv1d, nn.Conv2d],
+    pool: Union[nn.MaxPool1d, nn.MaxPool2d],
+) -> int:
+    """Returns new output size after applying a given convolution and pooling.
+
+    Assumes quadratic input dimensions of the data and the applied kernels, e.g.,
+    input_dim refers to data of shape (input_dim, input_dim) and all convolutions should
```

Comment:
I'm wondering if quadratic is too strong of an assumption. It should be easy to make it non-quadratic, right?

Reply:
then `get_new_cnn_output_size` could take a tuple and return a tuple

Reply:
we use quadratic kernerls throughout, so there is no difference in the dimensions. we could consider accepting rectangular input images as well, but this would add a couple of if-cases. I will have a look into how many...